### PR TITLE
feat(ast-spec): remove deprecated type params

### LINKED ISSUE #8927: Enhancement: Remove AST properties deprecated in v6
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

#6274 added renamed several AST properties and marked the old names as `@deprecated`. It'd be a major breaking change to remove them.

Let's do that for v8!

For example:

https://github.com/typescript-eslint/typescript-eslint/blob/df131e258c93e5714c88c0373cfeb2e1e75afc75/packages/ast-spec/src/base/ClassBase.ts#L56-L59

### Additional Info

We'll want to update the FAQ mention of these too.

https://github.com/typescript-eslint/typescript-eslint/blob/7c77b1b123d13fb585d588a9dd19f810c973a42a/docs/troubleshooting/FAQ.mdx#L470

💖

### PR DESCRIPTION
**Pull request by abrahamguo**
BREAKING CHANGE:
Removes properties from the AST.

## PR Checklist

- [x] Addresses an existing open issue: fixes #8927 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview
Remove @deprecated `typeParameters` properties that were deprecated in #6274

### CODE REVIEW

File: packages/typescript-estree/tests/lib/convert.test.ts
Comment (JoshuaKGoldberg) [inline packages/typescript-estree/tests/lib/convert.test.ts:1]: @abrahamguo it looks like there are some test failures, is that something you need help with?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
(waiting on test failures)

File: packages/typescript-estree/tests/lib/convert.test.ts
Comment (abrahamguo) [reply packages/typescript-estree/tests/lib/convert.test.ts:1]: I've been procrastinating investigating this deeply haha. I changed the tests to look at a different deprecated property but I'm unsure why they are now saying that no deprecation warnings were triggered. If you have any hunches, lmk!

Comment (abrahamguo) [reply packages/typescript-estree/tests/lib/convert.test.ts:1]: Actually, upon further inspection, it looks like the only other usages of `#withDeprecatedAliasGetter` are on `assertions` <=> `attributes`, but they have `suppressWarnings` set  to `true`, so they do not emit warnings. Therefore, the warnings created by `#withDeprecatedAliasGetter` are currently unused. Let me know how you would like to update (or remove) the tests.

Comment (abrahamguo) [reply packages/typescript-estree/tests/lib/convert.test.ts:1]: For now, I've simply removed the tests relating to printing warnings. Let me know if you'd prefer something different.

**Comment by abrahamguo** [comment]
FYI: This PR introduces new incorrect lint errors from `eslint-plugin-react/prop-types`, because it appears that `eslint-plugin-react` is reading from `typeParameters`, now removed in this PR.

File: packages/typescript-estree/tests/lib/convert.test.ts
Comment (JoshuaKGoldberg) [reply packages/typescript-estree/tests/lib/convert.test.ts:1]: I also just noticed this PR is against the `main` branch, but it should be against `v8`. I merged #8920 into `v8` yesterday that I think adds back in some logging stuff. Can you try switching to the now-deprecated `TSEnumDeclaration` `members`?

If there wasn't any use for the old code paths, I'd say they should just be deleted as dead code. But with #8920 I don't think that's the case here.

File: eslint.config.mjs [hunk 1]
```diff
@@ -513,6 +513,10 @@ export default tseslint.config(
       'react/jsx-no-target-blank': 'off',
       'react/no-unescaped-entities': 'off',
       'react-hooks/exhaustive-deps': 'warn', // TODO: enable it later
+      /* This rule
+        (1) causes false reports because it reads from `typeParameters` rather than `typeArguments`, and
+        (2) is unnecessary in TypeScript components */
```
Comment (JoshuaKGoldberg) [inline eslint.config.mjs:518]: For (1), if it's a transient problem, we'd want to link to the upstream issue (https://github.com/jsx-eslint/eslint-plugin-react/issues/3602). Vague comments tend to get out of date and don't help new readers understand the issue well.

For (2), yup, that's going to last beyond (1) - so I'd say it makes sense to disable the rule here always anyway.

```suggestion
```

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM, thanks! 🙌 Very excited to see most of our `@deprecated` AST spec properties gone.

Since this touches sensitive code, I'll leave it with `1 approval` for another member of @typescript-eslint/triage-team to take a look. I don't trust myself. 🙂 


Old timey Smurf cartoon showing a smurf sweeping a floor

**Pull request merged**